remoting/zookeeper/listener.go

```diff
@@ -63,16 +63,13 @@ func NewZkEventListener(client *gxzookeeper.ZookeeperClient) *ZkEventListener {
 
 // ListenServiceNodeEvent listen a path node event
 func (l *ZkEventListener) ListenServiceNodeEvent(zkPath string, listener remoting.DataListener) {
-	l.wg.Add(1)
-	go func(zkPath string, listener remoting.DataListener) {
-		if l.listenServiceNodeEvent(zkPath, listener) {
-			listener.DataChange(remoting.Event{Path: zkPath, Action: remoting.EventTypeDel})
-			l.pathMapLock.Lock()
-			delete(l.pathMap, zkPath)
-			l.pathMapLock.Unlock()
-		}
-		logger.Warnf("ListenServiceNodeEvent->listenSelf(zk path{%s}) goroutine exit now", zkPath)
-	}(zkPath, listener)
+	if l.listenServiceNodeEvent(zkPath, listener) {
```

Comment:
If `go func` is removed from here, then the code will be synced and is blocked until `l.listenServiceNodeEvent` returns. Why should it be removed?

Reply:
When calling this function, it is started asynchronously, so there is no problem with always posting